### ISSUE TITLE
Problem when Compiling for non Cmake Build System

### ISSUE THREAD

**Issue by X52p** [DiligentGraphics/DiligentCore#89]
I try to compile Diligent for a non Cmake build System (Scons). To test it, i wanted to build the Tutorial00_HelloLinux but with Scons instead of CMake. (Curently I'm working on Linux, but I want to Compile it on Windows too, on a later point)

Therefore i copied the .cpp file to a seperat folder and compiled Diligent as discriped in the Section "Your Project Does Not Use Cmake"

I tryed the following steps:
in the cloned repository (i call the folder DiligentFolder in the rest of this issiue):
cmake -H. -B./cmk_build/Linux64 -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Debug" -D CMAKE_INSTALL_PREFIX=install
and after than:
cmake --build . --target install
in /cmk_build/Linux64

I than use 
```'DiligentFolder/DiligentEngine/install/DiligentCore/headers'```
as my CPPPATH and
```'DiligentFolder/DiligentEngine/install/DiligentCore/lib'```
as my LIBPATH
i link against the following librarys:
```'libLinuxPlatform', 'GL', 'X11', 'xcb', 'GraphicsEngineOpenGL-static', 'GraphicsEngineVk-static', 'glslang', 'HLSL', 'OGLCompiler', 'OSDependent', 'SPIRVCross', 'SPIRV', 'SPIRV-Tools-opt', 'SPIRV-Tools', 'glew-static', 'vulkan'```

I get a lot of errors, but they all are similar to this:
```
/usr/bin/ld: 
DiligentFolder/DiligentEngine/install/DiligentCore/lib/libGraphicsEngineOpenGL-static.a(ShaderResourceBinding
GLImpl.cpp.o): in function `Diligent::RefCountedObject<Diligent::IShaderResourceBindingGL>::GetReferenceCounters() const':
DiligentFolder/DiligentEngine/DiligentCore/Common/interface/RefCountedObjectImpl.h:521: undefined reference to `DebugAsser
tionFailed(char const*, char const*, char const*, int)'
```

I propably did something wrong when compiling, but I don't know wich command is missing and couldn't find anything in the documentation.

**Comment by DiligentGraphics** [comment]
On Linux, the order in which you list libraries is important. Try to move `'libLinuxPlatform', 'GL', 'X11', 'xcb'` to the end of the list.
Also, try linking against `DiligentCore.lib` instead of `GraphicsEngineOpenGL-static` or `GraphicsEngineVk-static` because the library has all required dependencies (except for third-party). The easiest way though is to link with dynamic libraries (`GraphicsEngineOpenGL-shared` or `GraphicsEngineVk-shared`) as you will not need to sort out dependencies.

**Comment by X52p** [comment]
there is no DiligentCore.lib in the lib folder.

**Comment by DiligentGraphics** [comment]
It should be in `install/DiligentCore/lib/libDiligentCore.a`.
I checked the [log of the last Linux build](https://travis-ci.org/DiligentGraphics/DiligentCore/jobs/538058523#L1187) and the library should be there.

**Comment by X52p** [comment]
the folder only contains the following files:
libBasicPlatform.a
libCommon.a
libglew-static.a
libglslang.a
libGLSLTools.a
libGraphicsAccessories.a
libGraphicsEngine.a
libGraphicsEngineNextGenBase.a
libGraphicsEngineOpenGL.so
libGraphicsEngineOpenGL-static.a
libGraphicsEngineVk.so
libGraphicsEngineVk-static.a
libGraphicsTools.a
libHLSL2GLSLConverterLib.a
libHLSL.a
libLinuxPlatform.a
libOGLCompiler.a
libOSDependent.a
libPrimitives.a
libSPIRV.a
libSPIRVCross.a
libSPIRV-Tools.a
libSPIRV-Tools-opt.a

are the comands I discribed in my initial post corect or am I missing something?

**Comment by DiligentGraphics** [comment]
Yes, these commands seem to be correct. [These are commands](https://github.com/DiligentGraphics/DiligentCore/blob/master/.travis.yml#L94) run by the CI server and [here is the library](https://travis-ci.org/DiligentGraphics/DiligentCore/jobs/538058523#L1187) installed by the server.
Does it produce the `libDiligentCore.a` at all?

**Comment by X52p** [comment]
no, its not in the folder and I couldn't find it anyware else.
